### PR TITLE
TLF-101 - Submission page details 

### PR DESCRIPTION
**Pull request by vinodhasamiyappanHO**
### What
TLF-101 - COA - Added submission page details [TLF-101](https://collaboration.homeoffice.gov.uk/jira/browse/TLF-101)

### Why
- Adding submission page details with Unique Reference Number (6 Char) and with the changed address details.

### How
- Used crypto-random-string npm package to generate random string with custom characters to use Uppercase English Alphabets which excludes (I,L,O,Q,S,Z) and Number from 0 to 9. The [DEC alphabet](http://en.wikipedia.org/wiki/Alphanumeric) is an example of where these letters are excluded.
- Addred new data-sections.js to display only changed address sections. 
  - apps/coa/sections/changeDetails-data-sections.js
- Added related css
- Clearing session from the locals behaviour.

### Testing
- [x] Local testing passed
- [x] Test Lint Passed in Local

### CODE REVIEW

File: apps/coa/sections/summary-data-sections.js
Comment (vinodhasamiyappanHO) [inline apps/coa/sections/summary-data-sections.js:1]: The postal address field has been moved below the home address.

File: apps/coa/sections/changeDetails-data-sections.js
Comment (dk4g) [inline apps/coa/sections/changeDetails-data-sections.js:1]: file naming for consistency `change-details-data-sections.js`

File: apps/coa/sections/changeDetails-data-sections.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module.exports = {
+  'Home address details': {
```
Comment (dk4g) [inline apps/coa/sections/changeDetails-data-sections.js:2]: naming-convention for object keys would be more convenient to use later for multi-language translations, for example.

**Review by dk4g — CHANGES_REQUESTED** [review]

**Review by MikeMaratheftis — APPROVED** [review]
Code looks solid! Added validation for OISC or SRA numbers, tweaked error messages, and updated translations. Couldn't find any areas to improve or fix. Great job!

File: apps/coa/behaviours/submit-notify.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = superclass => class extends superclass {
+  async successHandler(req, res, next) {
+    try {
+	  const crs = await import('crypto-random-string');
+	  const uniqueRefNumber = crs.default({ length: 6, characters: 'ABCDEFGHJKMNPRTUVWXY0123456789' });
+	  req.sessionModel.set('uniqueRefNumber', uniqueRefNumber);
+    } catch (error) {
+	  return next(error);
```
[the remark this answers is not in the document]
Comment (vinodhasamiyappanHO) [reply apps/coa/behaviours/submit-notify.js]: Added the error log with the specific message and return the error with a message, which displays in the error page

File: apps/coa/sections/changeDetails-data-sections.js
Comment (vinodhasamiyappanHO) [reply apps/coa/sections/changeDetails-data-sections.js:1]: changes done

File: apps/coa/sections/changeDetails-data-sections.js (hunk 1 above)
Comment (vinodhasamiyappanHO) [reply apps/coa/sections/changeDetails-data-sections.js]: changes done

File: apps/coa/views/partials/summary-table.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  {{^omitFromSummary}}
+    <div class="confirm-row">
+      {{#isSeparator}}
+      {{/isSeparator}}
+      {{^isSeparator}}
+        <dt class="confirm-label">{{label}}</dt>
+        <dd class="confirm-value" data-value="{{value}}">{{value}}</dd>
+      {{^omitChangeLink}}
+        {{#changeLink}}
+          <dd><span><a class="govuk-link" href="{{changeLink}}" id="{{field}}-change-{{index}}">{{#t}}buttons.change{{/t}} <span class="visuallyhidden">{{changeLinkDescription}} from {{value}}</span></a></span></dd>
```
Comment (dk4g) [inline apps/coa/views/partials/summary-table.html:15]: couldn't see a translation for `changeLinkDescription`, is there a default value?

Comment (vinodhasamiyappanHO) [reply apps/coa/views/partials/summary-table.html:15]: The `changelinkdescription` is defined in hof/components/summary/index.js dynamically like below
 The value from the translation for summary sections label is assigned to the variable `changeLinkDescription`
` changeLinkDescription: this.translateChangeLink(inner.field, req),`
```
  translateChangeLink(key, req) {
    return req.translate([fields.${key}.changeLinkDescription,
      pages.confirm.fields.${key}.label,
      fields.${key}.summary,
      fields.${key}.label,
      fields.${key}.legend]);
  }
```

**Pull request merged**